src/sentry/models/widget.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import, print_function
+
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.constants import ObjectStatus
+from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+
+
+class WidgetDisplayTypes(object):
+    LINE_CHART = 0
+    AREA_CHART = 1
+    STACKED_AREA_CHART = 2
+    BAR_CHART = 3
+    PIE_CHART = 4
+    TABLE = 5
+    WORLD_MAP = 6
+    PERCENTAGE_AREA_CHART = 7
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.LINE_CHART, 'line'),
+            (cls.AREA_CHART, 'area'),
+            (cls.STACKED_AREA_CHART, 'stacked_area'),
+            (cls.BAR_CHART, 'bar'),
+            (cls.PIE_CHART, 'pie'),
+            (cls.TABLE, 'table'),
+            (cls.WORLD_MAP, 'world_map'),
+            (cls.PERCENTAGE_AREA_CHART, 'percentage_area_chart')
+        ]
+
+
+class WidgetDataSourceTypes(object):
+    DISCOVER_SAVED_SEARCH = 0
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.DISCOVER_SAVED_SEARCH, 'discover_saved_search'),
+        ]
+
+
+class WidgetDataSource(Model):
```

Comment:
@billyvg is this something that's needed? Is it common among the data sources to have an axis defined or is this something you'd lump in with display_options?

Reply:
Good point about the ordering, we should add it because it can affect things like chart color.

I was planning on handling the yAxis mapping using widget source ids inside of `display_options`